### PR TITLE
[90] update billing-ui packages

### LINKED ISSUE #70: Billing UI CSV Export Missing Columns
Billing UI CSV export missing these columns:

- Period
- Total

Note that the Period only shows up if the user clicks the 'Period' tab in the UI, but this doesn't make sense.  The Period should always be included.  Additionally, the Period shows up as 2 columns, 'fromDate' and 'toDate' (which is OK, this is just an FYI).

See attached example (CSV converted to XLSX) - UI has the columns, but file does not:

image

[billing-ui-sample-export.xlsx](https://github.com/overture-stack/billing/files/3848955/billing-ui-sample-export.xlsx)

Here is what the fromDate and toDate look like in the CSV if Period tab is clicked:

### LINKED ISSUE #90: [Security Vulnerability] Upgrade Multiple Packages
There are several security vulnerabilities to be addressed that have been opened in the past 1-2 months for Enrolment.

Specific Alerts below, please connect the individual PRs to this issue as required:

(1) https://github.com/overture-stack/billing/network/alert/billing-ui/package-lock.json/kind-of/open

(2) https://github.com/overture-stack/billing/network/alert/billing-ui/package-lock.json/acorn/open

(3) https://github.com/overture-stack/billing/network/alert/billing-invoice/yarn.lock/minimist/open

(4) https://github.com/overture-stack/billing/network/alert/billing-ui/package-lock.json/minimist/open

(5) https://github.com/overture-stack/billing/network/alert/invoice-api/yarn.lock/minimist/open

### PR DESCRIPTION
**Pull request by caravinci**
Closes #90. Supersedes #87, #91, #93, #95, #97, #99 and #100.

Also closes #70, by virtue of the necessary refactor from updating `react-bootstrap-table` to `react-bootstrap-table-next`.

### CODE REVIEW

File: billing-ui/src/services/projects/fetchProjects.js [hunk 1]
```diff
@@ -25,8 +25,11 @@ async function fetchProjects() {
     });
     const responseData = await response.json();
     user.token = response.headers.get('authorization');
+
     if ([401, 404].includes(response.status)) user.logout();
```
Comment (caravinci) [inline billing-ui/src/services/projects/fetchProjects.js:29]:
```suggestion
    if ([401].includes(response.status)) user.logout();
    // if ([401, 404].includes(response.status)) user.logout();
```

**Comment by caravinci** [comment]
Will add comments per file to ease reviewing

File: billing-ui/src/utils/formats.js [hunk 2]
```diff
@@ -15,8 +15,22 @@
  * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-export const formatCurrency = n => (n && typeof parseInt(n, 10) === 'number'
-    ? `$${n.toLocaleString(undefined, { minimumFractionDigits: 2 })}`
-    : '');
+export const formatCurrency = n => (
+  Number.isNaN(Number(n))
```
Comment (caravinci) [inline billing-ui/src/utils/formats.js:19]: handles 0 as numbers, even if falsy

File: billing-ui/src/utils/formats.js [hunk 3]
```diff
@@ -15,8 +15,22 @@
...
+export const formatNumber = n => (
+    Number.isNaN(Number(n))
+      ? ''
+      : n.toLocaleString()
+);
+
+export const formatPercentage = n => (
+    Number.isNaN(Number(n))
+      ? ''
+      : Number(n)
```
Comment (caravinci) [inline billing-ui/src/utils/formats.js:33]: NaN is being handled... but `0` vs `null` vs `undefined`?

File: billing-ui/src/utils/transforms.js [hunk 4]
```diff
@@ -33,6 +33,8 @@ export const collectionFlattener = (item, parnetKey, stem = '') => {
         : { [newStem]: item };
 };
 
-export const customNumberSort = (a, b, order, field) => ((order === 'desc')
-    ? a[field] - b[field]
-    : b[field] - a[field]);
+export const customNumberSort = (a, b, order) => (
```
Comment (caravinci) [inline billing-ui/src/utils/transforms.js:36]: The latest Bootstrap Tablas pass the value directly, not the row + fieldName anymore.

File: invoice-api/src/service/freshbooks.ts [hunk 5]
```diff
@@ -592,9 +592,12 @@ class FreshbooksService {
         const invoices = await this.getInvoicesListPaged(1, minDate, maxDate, allCustomerIDs);
 
         if (invoices.pages > 1) {
-            const pagedInvoices = await Promise.all(_.range(1, invoices.pages).map(i =>
-                this.getInvoicesListPaged(i, minDate, maxDate, allCustomerIDs)));
-            invoices.invoices = invoices.invoices.concat(_.flatten(pagedInvoices.map(item => item.invoices)));
+            const pagedInvoices = await Promise.all(_.range(1, invoices.pages)
+                .map(i => this.getInvoicesListPaged(i, minDate, maxDate, allCustomerIDs)));
```
Comment (caravinci) [inline invoice-api/src/service/freshbooks.ts:596]: just improving readability, and adding a log message, for ease of debugging.

File: billing-ui/package.json [hunk 6]
```diff
@@ -40,14 +40,14 @@
...
+        "filesize": "6.1.0",
+        "fs-extra": "9.0.1",
+        "gzip-size": "5.1.1",
         "html-webpack-plugin": "^4.4.1",
         "json-loader": "0.5.7",
-        "mini-css-extract-plugin": "1.3.0",
+        "mini-css-extract-plugin": "1.3.1",
         "node-sass": "^5.0.0",
-        "opn": "4.0.2",
+        "opn": "6.0.0",
```
Comment (ciaranschutte) [inline billing-ui/package.json:50]: deprecated package https://www.npmjs.com/package/opn

Comment (caravinci) [reply billing-ui/package.json:50]: OOH nice! let me switch that, see how it works

**Comment by ciaranschutte** [comment]
only other suggestion is remove moment package completely seen as you installed day as a lightweight alternative i think?
Looks great @caravinci !

**Comment by ciaranschutte** [comment]
> only other suggestion is remove moment package completely seen as you installed day as a lightweight alternative i think?
> Looks great @caravinci !

from offline: AntD relies on moment objects. replacing later

**Review by ciaranschutte — APPROVED** [review]

**Comment by caravinci** [comment]
For reference, in regards to Moment, and why I'm starting to replace it with Day.JS: 
> [We now generally consider Moment to be a legacy project in maintenance mode. It is not dead, but it is indeed done.](https://momentjs.com/docs/)

Hoping AntD will either remove this dependency, or we could consider moving away from using AntD altogether.

**Pull request merged**